File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String dividor = ":";
        String[] parts = logLine.split(dividor);

        return parts[1].trim();

    }

    public static String logLevel(String logLine) {
        String dividor = ":";
        String[] parts = logLine.split(dividor);

        return parts[0].replace("[","").replace("]","").toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
        
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

using std::string;

namespace log_line {
      string message(std::string line) {
         int colonIndex = line.find(":");
         return line.substr(colonIndex + 2);
}

    string log_level(string line) {
        int openBracketIndex = line.find("[");
        int closedBracketIndex = line.find("]");
        return line.substr(openBracketIndex + 1,(closedBracketIndex - openBracketIndex)-1);

}

    string reformat(string line) {
        // return the reformatted message
        return message(line) + " (" + log_level(line) + ")";
       }
    }// namespace log_line
